yt/yt/server/node/exec_node/job.cpp

```diff
@@ -4116,6 +4160,57 @@ void TJob::UpdateIOStatistics(const TStatistics& statistics)
     IORequestsWritten_ = newIORequestsWritten;
 }
 
+TDuration TJob::GetRunningPhaseDuration() const
+{
+    YT_ASSERT_THREAD_AFFINITY(JobThread);
+
+    if (RunningPhaseStartTime_ == TInstant::Zero()) {
+        return TDuration::Zero();
+    }
+
+    auto stopTime = RunningPhaseStopTime_;
+    if (stopTime == TInstant::Zero()) {
+        stopTime = TInstant::Now();
+    }
+
+    return stopTime - RunningPhaseStartTime_;
+}
+
+TResourceConsumption TJob::UpdateConsumptionTracking(std::optional<NClusterNode::TJobResources> newUsage)
+{
+    YT_ASSERT_THREAD_AFFINITY(JobThread);
+
+    if (RunningPhaseStartTime_ == TInstant::Zero()) {
+        if (newUsage) {
+            RunningPhaseResourceUsage_ = *newUsage;
```

Comment:
Why do you want to store this resource usage separately? Don't we already store current resource usage somewhere else?

Reply:
We don't store it, we update `ResourceHolder_->SetBaseResourceUsage(newUsage);`, and this `ResourceHolder` is from allocation and doesn't have "GetBaseResourceUsage" (only total with some strange Additional, which seems unused)